experimental/overhead_matching/kimera_spectacular_data_provider_test.cc

```diff
@@ -0,0 +1,178 @@
+    // check image
+    cv::Mat grey_image;
+    if (spec_bgr.channels() > 1) {
+        cv::cvtColor(spec_bgr, grey_image, cv::COLOR_BGR2GRAY);
+    } else {
+        grey_image = spec_bgr;
+    }
+    if (!common::video::images_equal(kimera_frame->img_, grey_image)) {
+        return false;
+    }
+    // check timestamps
+    if (kimera_frame->timestamp_ != spec_time.time_since_epoch().count()) {
+        return false;
+    }
+    // check index number
+    if (kimera_frame->id_ != sequence_index) {
+        return false;
+    }
+    return true;
+}
+bool compare_depth_frame(const std::unique_ptr<VIO::DepthFrame>& kimera_depth,
+                         const cv::Mat& spec_depth, const time::RobotTimestamp& spec_time,
+                         const int& sequence_index) {
+    // check image
+    if (!common::video::images_equal(kimera_depth->depth_img_, spec_depth)) {
+        return false;
+    }
+    // check timestamps
+    if (kimera_depth->timestamp_ != spec_time.time_since_epoch().count()) {
+        std::cout << "ID mismatch! " << kimera_depth->id_ << " "
+                  << spec_time.time_since_epoch().count() << std::endl;
+        return false;
+    }
+    // check index number
+    if (kimera_depth->id_ != sequence_index) {
+        return false;
+    }
+    return true;
+}
+
+TEST(KimeraSpectacularDataProviderTest, happy_case) {
+    // Setup
+    const std::filesystem::path log_path(
+        "external/spectacular_log_snippet/recording_2024-11-21_13-36-30");
+    SpectacularLog log(log_path);
+
+    // get initial IMU information
+    std::vector<double> times;
+    std::vector<robot::experimental::overhead_matching::ImuSample> original_imu_samples;
+    for (const double& t : log.accel_spline().ts()) {
+        time::RobotTimestamp robot_time = time::RobotTimestamp() + time::as_duration(t);
+        if (robot_time < log.min_imu_time() || robot_time > log.max_imu_time()) {
+            continue;
+        }
+        times.push_back(t);
+        auto maybe_sample = log.get_imu_sample(robot_time);
+        EXPECT_TRUE(maybe_sample.has_value());
+        original_imu_samples.push_back(maybe_sample.value());
+    }
+
+    const std::filesystem::path vio_config_path("");  // loads default params
+    VIO::VioParams vio_params(vio_config_path);
+    vio_params.parallel_run_ = false;
+
+    SpectacularDataProviderInterface s_interface(log_path, 0, std::numeric_limits<int>::max(),
+                                                 vio_params);
+
+    std::vector<VIO::ImuMeasurement> imu_queue;
+    std::vector<std::unique_ptr<VIO::Frame>> bgr_queue;
+    std::vector<std::unique_ptr<VIO::DepthFrame>> depth_queue;
+
+    auto imu_callback = [&imu_queue](const VIO::ImuMeasurement& measurement) -> void {
+        imu_queue.push_back(measurement);
+        return;
+    };
+
+    auto color_camera_callback = [&bgr_queue](std::unique_ptr<VIO::Frame>&& bgr_frame) -> void {
```

Comment:
I don't think this needs to be an r-value ref: https://stackoverflow.com/questions/37935393/pass-by-value-vs-pass-by-rvalue-reference